### PR TITLE
Only emit beat when the player is playing

### PR DESCRIPTION
**Pull request by Mecharyry**
## Problem
We've seen in a client application that it is possible for an `onBeat` to be emitted when the player is no longer playing.

## Solution
As the `Heart` uses a `Handler` we should check that the player is indeed playing in the `Heart.Heartbeat` before emitting an `onBeat`. 

### Test(s) added 
Yes, tested the `Heart` class.

### CODE REVIEW

File: core/src/main/java/com/novoda/noplayer/internal/Heart.java [hunk 1]
```diff
@@ -78,7 +78,9 @@ public Heartbeat(NoPlayer.HeartbeatCallback callback, NoPlayer player) {
 
         @Override
         public void run() {
-            callback.onBeat(player);
+            if (player.isPlaying()) {
```
Comment (rock3r) [inline core/src/main/java/com/novoda/noplayer/internal/Heart.java]: Why not starting/stopping the heartbeat when the playback starts/end instead? Seems more straightforward

Comment (Mecharyry) [reply core/src/main/java/com/novoda/noplayer/internal/Heart.java]: We do that already. The problem is that this handler might execute between player state changes 😢 it's kind of an edge case but we have seen it.

Comment (ouchadam) [reply core/src/main/java/com/novoda/noplayer/internal/Heart.java]: do you think we need to log/call a listener to let clients know this heartbeat has been skipped?

Comment (Mecharyry) [reply core/src/main/java/com/novoda/noplayer/internal/Heart.java]: I'm not sure tbh, I think emitting when it isn't playing is an error, it shouldn't happen. We would just be telling clients about behaviour we would never expect them to deal with. What do you think?

Comment (ouchadam) [reply core/src/main/java/com/novoda/noplayer/internal/Heart.java]: I agree, just a little paranoid that we might have changed behaviour (by fixing bad state).

I'd be happy with release notes that explain the scenario

File: core/src/test/java/com/novoda/noplayer/internal/HeartTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import org.mockito.stubbing.Answer;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.then;
+import static org.mockito.BDDMockito.will;
+import static org.mockito.Mockito.mock;
+
+public class HeartTest {
```
Comment (ouchadam) [inline core/src/test/java/com/novoda/noplayer/internal/HeartTest.java]: :100:

**Comment by Mecharyry** [comment]
Happy to merge @rock3r ?

**Comment by rock3r** [comment]
Yep works for me, if it's ok for Adam who knows the ins and outs :)

**Pull request merged**